access.lua

```diff
@@ -304,45 +281,28 @@ function serveYnhpanel()
     scandir("/usr/share/ssowat/portal/assets/themes/"..conf.theme, serveThemeFile)
 end
 
---
--- 6. Unprotected URLs
---
--- If the URL matches one of the `unprotected_urls` in the configuration file,
--- it means that the URL should not be protected by the SSO *but* headers have
--- to be sent if the user is already authenticated.
---
--- It means that you can let anyone access to an app, but if a user has already
--- been authenticated on the portal, he can have his authentication headers
--- passed to the app.
---
+local permission = hlp.get_best_permission()
 
-if longest_unprotected_match ~= ""
-and string.len(longest_unprotected_match) > string.len(longest_protected_match) then
-    if hlp.is_logged_in() then
+if permission then
```

Comment:
Thinking about this I think that we have some issue here.

The section `7. Basic HTTP Authentication` should be at my point of view before this, because here if the url match a permission not public and there are no user authenticated by a cookie the client is redirected ether if it is authenticated by a header.

I  also think that we need to manage the case if no permission match. Probably we just need to add this `hlp.redirect(conf.portal_url)`. This might avoid the actual loop.

Reply:
You're probably right, I don't know why that section was there and not before.

I fixed the redirect loop